src/main/java/javax/validation/Configuration.java

```diff
@@ -127,6 +129,19 @@
 	T clockProvider(ClockProvider clockProvider);
 
 	/**
+	 * Adds a value extractor. Has priority over any extractors for the same
+	 * type and type parameter detected through the service loader or given in
+	 * the XML configuration.
+	 *
+	 * @param extractor value extractor implementation
+	 * @return {@code this} following the chaining method pattern.
+	 * @throws IllegalArgumentException If more than one extractor for the same
```

Comment:
I don't think you want `IllegalArgumentException`, rather a specific BV exception. so that you can unify the exceptions coming from this method and from the bootstrap (XML + extractors).
On whether to throw an exception, I don't have a strong opinion but I'd tend to have it yes unless you can foresee that is limit proprietary extensions from providers.